barretenberg/cpp/src/barretenberg/ecc/groups/affine_element_impl.hpp

```diff
@@ -200,6 +200,25 @@ constexpr bool affine_element<Fq, Fr, T>::operator>(const affine_element& other)
     return false;
 }
 
+template <class Fq, class Fr, class T>
+constexpr std::optional<affine_element<Fq, Fr, T>> affine_element<Fq, Fr, T>::derive_from_x_coordinate(
+    const Fq& x, bool sign_bit) noexcept
+{
+    auto yy = x.sqr() * x + T::b;
+    if constexpr (T::has_a) {
+        yy += (x * T::a);
+    }
+    auto [found_root, y] = yy.sqrt();
+
+    if (found_root) {
+        if (uint256_t(y).get_bit(0) != sign_bit) {
+            y = -y;
+        }
```

Comment:
This is for determinism; a different sqrt algorithm could give -y instead of y and so this parity check allows all algorithms to get the "same" y

Reply:
I've copied your comment into the code as it helps describe what's going on here!